lib/licensed/dependency.rb

```diff
@@ -83,12 +83,16 @@ def notice_contents
          .grep(LEGAL_FILES_PATTERN)
          .select { |path| File.file?(path) }
          .sort # sorted by the path
-         .map { |path| { "sources" => normalize_source_path(path), "text" => File.read(path).rstrip } }
+         .map { |path| { "sources" => normalize_source_path(path), "text" => read_file_with_encoding_check(path) } }
          .select { |notice| notice["text"].length > 0 } # files with content only
     end
 
     private
 
+    def read_file_with_encoding_check(file_path)
```

Comment:
Is this something you've hit in the real world?  I'm curious what the scenario is that files contain invalid characters, and where `licensed` is raising errors.

Reply:
Sadly yes -- `AUTHORS` file for https://pypi.org/project/beautifulsoup4/ (version 4.7.1) is raising non-UTF-8 string exception.